app/models/king.rb

```diff
@@ -1,5 +1,6 @@
 class King < Piece
-    def legal (x, y)
+    def valid_move?(x, y)
+        return false unless super.valid_move?
```

Comment:
we also need to be adding is_obstructed? in these models here

Reply:
So this is where super comes in. the super valid_move method covers all of this:

`obstructed?(x,y) && off_the_board?(x,y) && in_check?(king) && not_moved_to_different_space?(x,y) && is_my_turn?`

so everything with the super.valid_move? line checks for all of those methods in the piece model. the only one that doesn't call on valid move is knight.rb because it doesn't follow the same obstruction rules, which is why it calls on all the other methods individually using super.